Documentation/signing-and-verification-guide.md

```diff
@@ -0,0 +1,265 @@
+* [Verifying Images with Rocket](#verifying-images-with-rocket)
+  * [Establishing Trust](#establishing-trust)
+  * [Example Usage](#example-usage)
+
+## Signing ACI Images
+
+### Generate a gpg signing key
+
+Create a file named `gpg-batch`
+
+```
+%echo Generating a default key
+Key-Type: default
+Subkey-Type: default
+Name-Real: Kelsey Hightower
+Name-Comment: ACI signing key
+Name-Email: kelsey.hightower@coreos.com
+Expire-Date: 0
+Passphrase: rocket
+%pubring rocket.pub
+%secring rocket.sec
+%commit
+%echo done
+```
+
+#### Generate the key using batch mode
+
+```
+$ gpg2 --batch --gen-key gpg-batch
+```
+
+#### List the keys
+
+```
+$ gpg2 --no-default-keyring --secret-keyring ./rocket.sec --keyring ./rocket.pub --list-keys
+```
+```
+./rocket.pub
+------------
+pub   2048R/26EF7A14 2015-01-09
+uid       [ unknown] Kelsey Hightower (ACI signing key) <kelsey.hightower@coreos.com>
+sub   2048R/B9C074CD 2015-01-09
+```
+
+#### Export the public key
+
+```
+$ gpg2 --no-default-keyring --armor \
+--secret-keyring ./rocket.sec --keyring ./rocket.pub \
+--output pubkeys.gpg \
+--export "<kelsey.hightower@coreos.com>"
+```
+
+### Sign the image
+
+```
+$ gpg2 --no-default-keyring --armor \
+--secret-keyring ./rocket.sec --keyring ./rocket.pub \
+--output hello-0.0.1-linux-amd64.sig \
+--detach-sig hello-0.0.1-linux-amd64.aci
+```
+
+#### Verify the image using gpg2
+
+```
+$ gpg2 --no-default-keyring \
+--secret-keyring ./rocket.sec --keyring ./rocket.pub \
+--verify hello-0.0.1-linux-amd64.sig hello-0.0.1-linux-amd64.aci
+```
+```
+gpg: Signature made Fri Jan  9 05:01:49 2015 PST using RSA key ID 26EF7A14
+gpg: Good signature from "Kelsey Hightower (ACI signing key) <kelsey.hightower@coreos.com>" [unknown]
+gpg: WARNING: This key is not certified with a trusted signature!
+gpg:          There is no indication that the signature belongs to the owner.
+Primary key fingerprint: B346 E31D E7E3 C6F9 D1D4  603F 4DFB 61BF 26EF 7A14
+```
+
+At this point you should have the following three files:
+
+* hello-0.0.1-linux-amd64.sig
+* hello-0.0.1-linux-amd64.aci
+* pubkeys.gpg
+
+## Distributing Images via Meta Discovery
+
+Host an HTML page with the following meta tags:
+
+```
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta name="ac-discovery" content="example.com/hello https://example.com/images/{name}-{version}-{os}-{arch}.{ext}">
+    <meta name="ac-discovery-pubkeys" content="example.com/hello https://example.com/pubkeys.gpg">
+  <head>
+<html>
+```
+
+Serve the following files at the locations described in the meta tags:
+
+* hello-0.0.1-linux-amd64.sig
+* hello-0.0.1-linux-amd64.aci
+* pubkeys.gpg
+
+### Rocket Integration
+
+Lets walk through the steps rocket takes when installing images using Meta Discovery.
+The following rocket command:
+
+```
+$ rkt run example.com/hello:0.0.1
+```
+
+translates into the following curl commands:
```

Comment:
I don't know if I like this curl translation since it's not quite accurate; how about "results in rocket retrieving the following URIs"